qiskit/validation/fields/custom.py

```diff
@@ -51,40 +51,72 @@ class InstructionParameter(ModelTypeValidator):
 
     This field provides support for parsing objects of types that uses by
     qobj.experiments.instructions.parameters:
-    * basic Python types: complex, int, float, str
-    * ``numpy``: integer, float
+    * basic Python types: complex, int, float, str, list
+    * ``numpy``: integer, float, ndarray
     * ``sympy``: Symbol, Basic
 
     Note that by using this field, serialization-deserialization round-tripping
     becomes not possible, as certain types serialize to the same Python basic
     type (for example, numpy.float and regular float). If possible, it is
     recommended that more specific and defined fields are used instead.
     """
+
     valid_types = (complex, int, float, str,
-                   numpy.integer, numpy.float, sympy.Basic, sympy.Symbol)
+                   numpy.integer, numpy.float, sympy.Basic, sympy.Symbol,
+                   list, numpy.ndarray)
+
+    default_error_messages = {
+        'invalid': '{input} cannot be parsed as a parameter.',
+        'format': '"{input}" cannot be formatted as a parameter.'
+    }
 
     def _serialize(self, value, attr, obj):
         # pylint: disable=too-many-return-statements
-        if isinstance(value, (float, int, str)):
-            return value
+        if is_collection(value):
+            return [self._serialize(item, attr, obj) for item in value]
+
```

Comment:
These changes seem to have come here from #1970.. can you do a rebase?

Reply:
yes